vispy/visuals/colorbar.py

```diff
@@ -679,5 +679,5 @@ def size(self, size):
         elif height == 0.:
             raise ValueError("height must be non-zero, not %s" % (height, ))
 
-        self._halfdim = (width, height)
+        self._halfdim = (width / 2, height / 2)
```

Comment:
I figured since `size` as a property is returned as `(halfw * 2., halfh * 2.)` (or `(halfh * 2., halfw * 2.)`), if you want odd pixel amounts or a size that isn't in pixels you need to assume that `width`/`height` are floats. It might actually be good to convert them to floats so that Python 2 doesn't break them when doing math.